src/main/java/core/basesyntax/dao/jdbc/DaoJdbcManufacturerImpl.java

```diff
@@ -0,0 +1,126 @@
+        String query = "SELECT * FROM manufacturers WHERE manufacturers_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            statement.close();
+            List<Manufacturer> list = new ArrayList<>();
+            while (resultSet.next()) {
+                list.add(parseResultSet(resultSet));
+            }
+            return list;
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("ERROR: getAll() method has failed", throwables);
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer item) {
+        if (get(item.getId()).isPresent()) {
+            String query = "UPDATE manufacturers SET manufacturer_name = ?, "
+                    + "manufacturer_country = ? WHERE manufacturer_id = ?";
+            try (Connection connection = ConnectionUtil.getConnection()) {
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setString(1, item.getName());
+                statement.setString(2, item.getCountry());
+                statement.setLong(3, item.getId());
+                statement.executeUpdate();
+                statement.close();
+                return get(item.getId()).get();
+            } catch (SQLException throwables) {
+                throw new DataProcessingException("ERROR: update() method has failed for "
+                        + item, throwables);
+            }
+        }
+        throw new RuntimeException("ERROR: Database contains no " + item + " object to update");
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        if (get(id).isPresent()) {
+            String query = "UPDATE manufacturers SET manufacturer_deleted = TRUE "
+                    + "WHERE manufacturer_id = ?";
+            try (Connection connection = ConnectionUtil.getConnection()) {
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setLong(1, id);
+                int result = statement.executeUpdate();
+                statement.close();
+                return result > 0;
+            } catch (SQLException throwables) {
+                throw new DataProcessingException("ERROR: delete() method has failed for id="
+                        + id, throwables);
+            }
+        }
+        throw new RuntimeException("ERROR: Database contains no objects with id=" + id);
+    }
+
+    private Manufacturer parseResultSet(ResultSet resultSet) throws SQLException {
+        long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String manufacturerName = resultSet.getObject("manufacturer_name", String.class);
+        String manufacturerCountry = resultSet.getObject("manufacturer_country", String.class);
```

Comment:
if you want you can use method `getString` too

Reply:
Because general purpose of this is to get null as an indicator for absence of object, right?